### DISCUSSION TITLE
Same output parameter from multiple plugins

### DISCUSSION THREAD

**Discussion by mr-loop-1** [Green-Software-Foundation/hack#146] [Ask Anything] [answered]
I am currently initiating a project around plugins, and came around this question,

What happens when two or more plugins are calculating the same metric, for ex. `cpu/energy`. I don't know if this is something very naive, but it could happen right? Naive because, if two plugins are calculating the same canonical unit, they should ideally give the same value.

**Comment by jawache** [comment, marked as answer]
What should happen is the plugin later on in the pipeline overrides the value output by the plugin previous.

It's valid that different plugins return different values for a parameter, they could be calculating them using different methodologies for instance. There are several examples already of plugins which calculate this and yet return very different values. The boavista CPU plug-in computes a very different CPU energy value because it's computing primary energy and is therefore computing a completely different methodology than the other plugins.

One use case for two plugins in the same pipeline is perhaps the earlier plug-in gives a more ubiquitous general value and a later plug-in only gives more accurate values for a specific subset of CPUs. The canonical example here might be a generic plugin which tries to calculate a rough CPU energy for all CPUs and then later on you add an Intel only plugin which perhaps would give you more accurate values just for an Intel CPU.

**Comment by mr-loop-1** [comment]
@jawache yes valid points. Maybe something like `@pluginName/cpu/energy` could be implemented at a later level for the same.
